### Laboratorio 6 - Ejercicio 1
Stefano Aragoni, Carol Arévalo, Luis Santos, Diego Perdomo

-----------

`Ejercicio 3`

Tiene la tarea de modelar el crecimiento de la población en múltiples escalas, desde organismos individuales hasta la población global. Para esto debe implementar un modelo multiescala en Python para analizar el crecimiento poblacional y su impacto

-----------

##### **Preguntas**



*a. ¿Qué es el modelado multiescala y por qué es esencial para comprender sistemas complejos como el crecimiento de la población?*

El modelado multiescala se refiere a la técnica de modelar un sistema en diferentes escalas, desde la más pequeña (micro) hasta la más grande (macro). Es una poderosa técnica computacional que nos permite simular y analizar sistemas complejos en diferentes niveles de detalle, desde la escala atómica y molecular, hasta el nivel macroscópico.

*obtenido de las presentación de Alberto



*b. Describe el modelo a nivel individual en esta simulación multiescala. ¿Qué factores se consideran a escala individual?*

Edad: La capacidad reproductiva varía con la edad.
Salud: La salud puede influir en la longevidad y la capacidad reproductiva.
Entorno: Acceso a recursos, calidad de vida, etc.
Tasa de reproducción: Número promedio de descendientes por individuo.

*c. Discuta el modelo a escala de comunidad local. ¿Cómo incorpora interacciones entre individuos y factores ambientales?*

A nivel de comunidad local, el modelo podría considerar:

Densidad poblacional: Afecta la competencia por recursos y puede influir en la tasa de reproducción.
Recursos disponibles: Alimentos, agua, refugio, etc.
Interacciones sociales: Como matrimonios, formación de familias, etc.
Factores ambientales: Como desastres naturales, enfermedades, etc.
Las interacciones entre individuos pueden modelarse a través de tasas de encuentro, formación de parejas y reproducción. Los factores ambientales pueden modelarse como eventos estocásticos que afectan la mortalidad o la natalidad.

*d. Explique el modelo a escala global. ¿Qué parámetros y factores se consideran al modelar toda la población mundial?*

A nivel global, el modelo podría considerar:

Migraciones: Movimientos de poblaciones entre regiones o países.
Políticas gubernamentales: Políticas de control de natalidad, salud pública, etc.
Catástrofes globales: Guerras, pandemias, grandes desastres naturales.
Tendencias económicas y tecnológicas: Que pueden influir en la calidad de vida y la tasa de reproducción.


-----------

##### **Modelación**

Modelo Multiescala Simple para el Crecimiento Poblacional:

Nivel individual: Cada individuo tiene una probabilidad de reproducción y una probabilidad de mortalidad basadas en su edad.

Nivel de comunidad local: La tasa de reproducción se ajusta según la densidad poblacional.

Nivel global: Se ajustan las tasas de crecimiento según las migraciones.

In [1]:
import random

class Individuo:
    def __init__(self, edad):
        self.edad = edad

    def reproducir(self):
        # Probabilidad de reproducción disminuye con la edad
        prob_reproduccion = max(0.2 - self.edad * 0.01, 0.05)
        return random.random() < prob_reproduccion

    def morir(self):
        # Probabilidad de mortalidad aumenta con la edad
        prob_mortalidad = self.edad * 0.02
        return random.random() < prob_mortalidad

    def envejecer(self):
        self.edad += 1

class Comunidad:
    def __init__(self, poblacion_inicial):
        self.individuos = [Individuo(random.randint(0, 50)) for _ in range(poblacion_inicial)]

    def paso_temporal(self):
        # Nivel individual
        nuevos_individuos = []
        for individuo in self.individuos:
            individuo.envejecer()
            if individuo.reproducir():
                nuevos_individuos.append(Individuo(0))
            if not individuo.morir():
                nuevos_individuos.append(individuo)
        self.individuos = nuevos_individuos

        # Nivel de comunidad local
        if len(self.individuos) > 1000:
            # Reducir la tasa de reproducción si la población es muy alta
            self.individuos = random.sample(self.individuos, int(len(self.individuos) * 0.9))

        # Nivel global (simplificado)
        # Supongamos que hay una migración neta de 10 individuos cada año
        migracion = 10
        self.individuos += [Individuo(random.randint(0, 50)) for _ in range(migracion)]

    def poblacion_total(self):
        return len(self.individuos)

# Simulación
comunidad = Comunidad(500)
for año in range(10):
    comunidad.paso_temporal()
    print(f"Año {año+1}: Población total = {comunidad.poblacion_total()}")


Año 1: Población total = 284
Año 2: Población total = 221
Año 3: Población total = 202
Año 4: Población total = 205
Año 5: Población total = 217
Año 6: Población total = 230
Año 7: Población total = 252
Año 8: Población total = 269
Año 9: Población total = 277
Año 10: Población total = 295


*Resumen y Análisis:*

Inicio de la Simulación: La población inicial era de 500 individuos.

Año 1: Se observa una disminución significativa en la población, llegando a 284 individuos. Esto puede ser el resultado de la mortalidad basada en la edad y la reducción de la tasa de reproducción debido a la alta densidad poblacional inicial.

Años 2 y 3: La población sigue disminuyendo, lo que indica que la tasa de mortalidad sigue siendo más alta que la tasa de reproducción, a pesar de la migración neta positiva.

Año 4: La población se estabiliza ligeramente, con un pequeño aumento. Esto puede ser el resultado de un equilibrio entre la tasa de reproducción y la tasa de mortalidad, junto con la migración.

Años 5 a 10: Se observa un crecimiento constante y gradual de la población. Esto sugiere que la comunidad ha alcanzado un equilibrio donde la tasa de reproducción, ajustada por la densidad poblacional y la migración, supera la tasa de mortalidad.

Tendencia General: Después de una disminución inicial en la población, la comunidad muestra signos de recuperación y crecimiento sostenido. Esto puede ser el resultado de la adaptación de la comunidad a las condiciones iniciales y la influencia de factores a nivel individual, local y global.